plugins/modules/hcloud_floating_ip.py

```diff
@@ -259,17 +259,24 @@ def _update_floating_ip(self):
                 self._mark_as_changed()
 
             server = self.module.params.get("server")
-            if server is not None:
-                if self.module.params.get("force") or self.hcloud_floating_ip.server is None:
+            if server is not None and self.hcloud_floating_ip.server is not None:
+                if self.module.params.get("force") and server != self.hcloud_floating_ip.server.name:
                     if not self.module.check_mode:
                         self.hcloud_floating_ip.assign(
                             self.client.servers.get_by_name(self.module.params.get("server"))
                         )
-                else:
+                        self._mark_as_changed()
+                elif server != self.hcloud_floating_ip.server.name:
                     self.module.warn(
-                        "Floating IP is already assigned to server %s. You need to unassign the Floating IP or use force=yes."
+                        "Floating IP is already assigned to another server %s. You need to unassign the Floating IP or use force=yes."
                         % self.hcloud_floating_ip.server.name
                     )
+                    self._mark_as_changed()
+            elif server is not None and self.hcloud_floating_ip.server is None:
+                if not self.module.check_mode:
+                    self.hcloud_floating_ip.assign(
+                        self.client.servers.get_by_name(self.module.params.get("server"))
```

Comment:
You can use `server` here instead of getting it again from the module parameters (should be done in line 266 too)

Reply:
Done